src/program.cpp

```diff
@@ -299,24 +292,24 @@ void program::compile(const std::vector<target>& targets, std::vector<compile_op
                 MIGRAPHX_THROW("Dangling reference in module " + current_mod->name() +
                                " from instruction " + std::to_string(index));
             }
-            current_mod->finalize(this->impl->contexts[root_target_id]);
         }
     }
+    this->finalize();
```

Comment:
This would only finalize main module.

Reply:
It will finalize all the submodules to main as well:

https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/module.cpp#L666